src/serial_stream.cpp

```diff
@@ -5,7 +5,7 @@
 
 namespace create {
 
-  SerialStream::SerialStream(boost::shared_ptr<Data> d, const uint8_t& header) : Serial(d), headerByte(header){
+  SerialStream::SerialStream(boost::shared_ptr<Data> d, const uint8_t& header) : Serial(d), headerByte(header), readState(READ_HEADER){
```

Comment:
nitpick: Better to initialize `readState` *before* `headerByte` so they are in the order they are defined in the class.